scripts/skills/special/rf_first_turn_initiative.nut

```diff
@@ -0,0 +1,77 @@
+this.rf_first_turn_initiative <- ::inherit("scripts/skills/skill", {
+	m = {
+		// Public
+		InitiativePerAP = 40.0,	// For every this much Initiative an entity gains 1 AP. The bonus can be negative
+		FlatAPBonus = -2,	// This is to balance out this new bonus for most entities
+
+		// Private
+		APBonus = 0
+
+		// Const
+		IconNegative = "ui/traits/trait_icon_25.png"	// Hesitant Trait
+		TotalMinimum = -3,
+		TotalMaximum = 7,	// Maximum is just incase so that enemies that accidentally have extreme initiative don't one-shot you Turn one
+	},
+	function create()
+	{
+		this.m.ID = "special.rf_first_turn_initiative";
+		this.m.Name = "First Turn Initiative";
+		this.m.Description = "During the first turn of every combat, this characters [Action Points|Concept.ActionPoints] are modified depending on their [Initiative|Concept.Initiative] at that point.";
+		this.m.Icon = "ui/traits/trait_icon_46.png";	// Impatient Trait icon by default
+		this.m.Type = ::Const.SkillType.StatusEffect;
+		this.m.Order = ::Const.SkillOrder.Last + 100;		// Some effects may set the Action Points to a flat value instead of increasing them, like 'possessed_undead_effect'
+		this.m.IsActive = false;
+		this.m.IsStacking = false;
+		this.m.IsSerialized = false;
+		this.m.IsRemovedAfterBattle = true;
+	}
+
+	function getDescription()
+	{
+		local description = this.m.Description;
+		description += "\nThis character gains 1 [Action Point|Concept.ActionPoints] for every " + ::MSU.Text.colorGreen(this.m.InitiativePerAP) + " [Initiative|Concept.Initiative] which is then modified by a flat " + ::MSU.Text.colorizeValue(this.m.FlatAPBonus) + " [Action Points|Concept.ActionPoints].";
+		return ::Reforged.Mod.Tooltips.parseString(description);
+	}
+
+	function getTooltip()
+	{
+		local tooltip = this.skill.getTooltip();
+		tooltip.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/action_points.png",
+			text = ::MSU.Text.colorizeValue(this.m.APBonus) + " Action Points for this turn"
+		});
+		return tooltip;
+	}
+
+	function isHidden()
+	{
+		return (this.m.APBonus == 0);
+	}
+
+	function onUpdate( _properties )
+	{
+		_properties.ActionPoints += this.m.APBonus;
+	}
+
+	function onNewRound()
```

Comment:
Wouldn't it be sufficient to do this in `onCombatStarted()`? Then you won't need the switcheroo and it can be much cleaner. Similarly you could add the effect in `actor.nut` during the `actor.onCombatStart` function before calling the original function.

Reply:
I don't see where the `onCombatStart` exists in the actor.nut. I looked in vanilla, msu and reforged. No one seems to declare it.

Reply:
I'm unsure if `onCombatStarted` from the skill.nut is the answer here. It looks to be called in the same way as `onNewRound` is. In both function the `IsUpdating` is set to true, which prevents manual update calls.

Reply:
Ah yes you are correct about `actor.nut` not having `onCombatStart`. In vanilla that only exists in `player.nut` which we, over at MSU, are "fixing" by adding it also to `actor.nut` because this is what causes `onCombatStarted` to not be called for non-player actors in vanilla which we consider a [bug](http://battlebrothersgame.com/forums/topic/oncombatstarted-is-not-called-for-ai-characters/).

Regarding the update, you are correct `IsUpdating` is true during `onCombatStarted`. However, this means that as long as `onNewRound` is called **after** `onCombatStarted` the character has already updated at the end of `onCombatStarted` and therefore the properties should be up to date. So you actually don't need to sneak in a new update here.